mysql/resource_ti_config_variable.go

```diff
@@ -0,0 +1,171 @@
+package mysql
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/creasty/defaults"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/tidwall/gjson"
+)
+
+func resourceTiConfigVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateOrUpdateConfigVariable,
+		Read:   ReadConfigVariable,
+		Update: CreateOrUpdateConfigVariable,
+		Delete: DeleteConfigVariable,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"pd", "tikv"}, true),
+			},
+			"instance": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func CreateOrUpdateConfigVariable(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*MySQLConfiguration).Db
+
+	varName := d.Get("name").(string)
+	varValue := d.Get("value").(string)
+	varInstanceType := d.Get("type").(string)
+	varInstance := d.Get("instance").(string)
+
+	var warnLevel, warnMessage string
+	var warnCode int = 0
+
+	configQuery := fmt.Sprintf("SET CONFIG %s %s=", varInstanceType, quoteIdentifier(varName))
+
+	if varInstance != "" {
+		configQuery = fmt.Sprintf("SET CONFIG \"%s\" %s=", varInstance, quoteIdentifier(varName))
+	}
+
+	if isNumeric(varValue) {
```

Comment:
This doesn't sound right. For DBs respecting data types, it depends on the type of key. Type of value may help (maybe that's your approximation), but it doesn't always work.

Doesn't TiDB respect it (like MySQL) in case you quote everything?

Reply:
L64: 
If instance is in place then it requires extra quoting like in the example 
`
set config "127.0.0.1:20180"  `split.qps-threshold`=1000
`

Everything else should be configured like this
`set config tikv `split.qps-threshold`=1000`

ref. https://docs.pingcap.com/tidb/stable/dynamic-config

Reply:
> L64: If instance is in place then it requires extra quoting like in the example ```set config "127.0.0.1:20180" `split.qps-threshold`=1000 ```
> 
> Everything else should be configured like this ```set config tikv `split.qps-threshold`=1000```

We don't understand each other - is it bad, if you quote it like this?
```
set config tikv `split.qps-threshold`='1000'
```
I believe it would be also acceptable and it would make it a bit easier to understand, but I'd leave it up to you.

Reply:
Got it 😀 Thank you 👍 
Remove obsolete code